src/main/java/com/javadeobfuscator/deobfuscator/rules/allatori/RuleStringDecryptor.java

```diff
@@ -48,44 +49,40 @@ public String test(Deobfuscator deobfuscator) {
             for (MethodNode methodNode : classNode.methods) {
                 Frame<SourceValue>[] frames = null;
                 for (AbstractInsnNode ain : TransformerHelper.instructionIterator(methodNode)) {
-                    if (ain.getOpcode() == Opcodes.INVOKESTATIC) {
-                        MethodInsnNode m = (MethodInsnNode) ain;
-                        String strCl = m.owner;
-                        if ((m.desc.equals("(Ljava/lang/Object;)Ljava/lang/String;") || m.desc.equals("(Ljava/lang/String;)Ljava/lang/String;"))
-                            && deobfuscator.getClasses().containsKey(strCl)) {
-                            ClassNode innerClassNode = deobfuscator.getClasses().get(strCl);
-                            MethodNode decryptorNode = innerClassNode.methods.stream()
-                                    .filter(mn -> mn.name.equals(m.name) && mn.desc.equals(m.desc))
-                                    .findFirst().orElse(null);
-                            if (decryptorNode == null || decryptorNode.instructions == null) {
-                                continue;
-                            }
-                            boolean isAllatori = true;
-
-                            isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "charAt", "(I)C");
-                            isAllatori = isAllatori && TransformerHelper.containsInvokeVirtual(decryptorNode, "java/lang/String", "length", "()I");
-                            isAllatori = isAllatori && TransformerHelper.countOccurencesOf(decryptorNode, IXOR) > 2;
-                            isAllatori = isAllatori && TransformerHelper.countOccurencesOf(decryptorNode, NEWARRAY) > 0;
+                    if (ain.getOpcode() != Opcodes.INVOKESTATIC) {
+                        continue;
+                    }
+                    MethodInsnNode m = (MethodInsnNode) ain;
+                    String strCl = m.owner;
+                    if ((!m.desc.equals("(Ljava/lang/Object;)Ljava/lang/String;") && !m.desc.equals("(Ljava/lang/String;)Ljava/lang/String;"))
+                        || !deobfuscator.getClasses().containsKey(strCl)) {
+                        continue;
+                    }
+                    ClassNode innerClassNode = deobfuscator.getClasses().get(strCl);
+                    MethodNode decryptorNode = innerClassNode.methods.stream()
+                            .filter(mn -> mn.name.equals(m.name) && mn.desc.equals(m.desc))
+                            .findFirst().orElse(null);
+                    if (decryptorNode == null || decryptorNode.instructions == null) {
+                        continue;
+                    }
 
-                            if (!isAllatori) {
-                                continue;
-                            }
+                    if (!com.javadeobfuscator.deobfuscator.transformers.allatori.StringEncryptionTransformer.isAllatoriMethod(decryptorNode)) {
```

Comment:
Perhaps these kinds of methods should be thrown in their own class or something.